src/pal/src/eventprovider/lttng/eventprovdotnetruntimerundown.cpp

```diff
@@ -0,0 +1,1576 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+/******************************************************************
+
+DO NOT MODIFY. AUTOGENERATED FILE. 
+This file is generated using the script from <root>/src/inc/genXplatEtw.pl
+
+******************************************************************/
+
+
+#define TRACEPOINT_CREATE_PROBES
+#define TRACEPOINT_DEFINE
+#include "lttng/tpdotnetruntimerundown.h"
+extern "C" BOOL  EventEnabledCLRStackWalkDCStart(){ return TRUE;}
+extern "C" ULONG  FireEtwCLRStackWalkDCStart(
+                    const unsigned short  ClrInstanceID,
+                    const unsigned char  Reserved1,
+                    const unsigned char  Reserved2,
+                    const unsigned int  FrameCount,
+                    const void ** Stack
+)
```

Comment:
Have you looked at the generated assembly, of both this function and the callers ? I'm worried that you might be taking a function call even though the tracepoint is not enabled.

This is why we have introduced the tracepoint_enabled() and the do_tracepoint() macros in lttng-ust 2.7: you can embed the do_tracepoint() macro within your function, but leave the tracepoint_enabled() check in the caller, before hitting the function call.